src/psyclone/psyir/nodes/reference.py

```diff
@@ -274,8 +285,33 @@ def is_full_range(self, index):
                 ScalarType.Intrinsic.INTEGER
                 and lower.children[1].value == str(index+1)):
```

Comment:
I think this would be a separate issue to this PR as both LBOUND/UBOUND and array indexing are pre-existing. However, I am comfortable with the implementation as it is. Yes, we kept the Fortran syntax for `lbound` and `ubound` but I would call this the array dimension not the index, and we start from 0 for array indices as we use children to capture an arrays indices (and Python starts from 0). I guess we could change lbound and ubound, but I think 'array dimension 1 having an index of 0' in the PSyIR is OK.